tests/utils/pod.go

```diff
@@ -0,0 +1,93 @@
+package utils
+
+import (
+	"time"
+
+	k8sv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	PodCreateTime  = 30 * time.Second
+	PodDeleteTime  = 30 * time.Second
+	PodWaitForTime = 30 * time.Second
+)
+
+// Create a Pod with the passed in PVC mounted under /pvc. You can then use the executor utilities to
+// run commands against the PVC through this Pod.
+func CreateExecutorPodWithPVC(clientSet *kubernetes.Clientset, podName, namespace string, pvc *k8sv1.PersistentVolumeClaim) (*k8sv1.Pod, error) {
+	var pod *k8sv1.Pod
+	podDef := newExecutorPodWithPVC(podName, "ls -1 /pvc | wc -l", pvc)
+	err := wait.PollImmediate(2*time.Second, PodCreateTime, func() (bool, error) {
+		var err error
+		pod, err = clientSet.CoreV1().Pods(namespace).Create(podDef)
+		if err != nil {
+			return false, err
+		}
+		return true, nil
+	})
+	return pod, err
+}
+
+// Delete the passed in Pod from the passed in Namespace
+func DeletePod(clientSet *kubernetes.Clientset, pod *k8sv1.Pod, namespace string) error {
+	return wait.PollImmediate(2*time.Second, PodDeleteTime, func() (bool, error) {
+		err := clientSet.CoreV1().Pods(namespace).Delete(pod.GetName(), &metav1.DeleteOptions{})
+		if err != nil {
+			return false, nil
+		}
+		return true, nil
+	})
+}
+
+func newExecutorPodWithPVC(podName, cmd string, pvc *k8sv1.PersistentVolumeClaim) *k8sv1.Pod {
```

Comment:
`cmd` appears to be ignored

Reply:
Correct, it is, we want this pod to be here so we can run our utility methods against it that can work with stdout/err/etc. The command simply needs to keep the pod alive long enough for us to connect to it.